### PR TITLE
ENG-860: Correct version suffixes for bin_tar

### PR DESCRIPTION
**Pull request by Sudokamikaze**

### CODE REVIEW

**Comment by Sudokamikaze** [comment]
Here's how it looks like on CentOS 8:
```
Percona-Server-5.7.29-32-Linux.x86_64.ssl111.readline7.tar.gz
```

Before it was: `Percona-Server-5.7.29-32-Linux.x86_64.ssl1:111.tar.gz`

File: build-ps/percona-server-5.7_builder.sh [hunk 1]
```diff
@@ -716,14 +716,25 @@ build_tarball(){
...
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F ":" '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            else
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            fi
         else
-            SSL_VER_TMP=$(dpkg -l|grep -i libssl|grep -v "libssl\-"|head -n1|awk '{print $2}'|awk -F ":" '{print $1}'|sed 's/libssl/ssl/g'|sed 's/\.//g')
-            export SSL_VER=".${SSL_VER_TMP}"
+            SSL_VER_TMP=$(dpkg -l|grep -i libssl|head -n1|awk '{print $3}'|awk -F "." '{print $1 $2}')
         fi
+        export SSL_VER=".ssh${SSL_VER_TMP}"
```
Comment (EvgeniyPatlan) [inline build-ps/percona-server-5.7_builder.sh]: ssh?

**Comment by Sudokamikaze** [comment]
And here's another one but for Ubuntu 20.04:
```
Percona-Server-5.7.29-32-Linux.x86_64.ssl11.readline8.tar.gz
```

File: build-ps/percona-server-5.7_builder.sh [hunk 2]
```diff
@@ -716,14 +716,25 @@ build_tarball(){
...
-            export SSL_VER=".ssl${SSL_VER_TMP}"
+            if [ "${RHEL}" -eq 8 ]; then
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F ":" '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            else
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            fi
         else
-            SSL_VER_TMP=$(dpkg -l|grep -i libssl|grep -v "libssl\-"|head -n1|awk '{print $2}'|awk -F ":" '{print $1}'|sed 's/libssl/ssl/g'|sed 's/\.//g')
-            export SSL_VER=".${SSL_VER_TMP}"
+            SSL_VER_TMP=$(dpkg -l|grep -i libssl|head -n1|awk '{print $3}'|awk -F "." '{print $1 $2}')
```
Comment (EvgeniyPatlan) [inline build-ps/percona-server-5.7_builder.sh]:
```
dpkg -l|grep -i libssl|head -n1|awk '{print $3}'
1.0.2n-1ubuntu5.3
```
and your command returns:
```
dpkg -l|grep -i libssl|head -n1|awk '{print $3}'|awk -F "." '{print $1 $2}'
10
```
previously it was ssl100 which is also incorrect as version was

```
dpkg -l|grep -i libssl|grep -v "libssl\-"| dpkg -l|grep -i libssl|grep -v "libssl\-"
ii  libssl1.0.0:amd64                               1.0.2n-1ubuntu5.3                                           amd64        Secure Sockets Layer toolkit - shared libraries
ii  libssl1.1:amd64                                 1.1.1-1ubuntu2.1~18.04.5                                    amd64        Secure Sockets Layer toolkit - shared libraries
```

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: build-ps/percona-server-5.7_builder.sh [hunk 3]
```diff
@@ -716,14 +716,26 @@ build_tarball(){
     export CXXFLAGS="${CFLAGS}"
     if [ "${YASSL}" = 0 ]; then
         if [ -f /etc/redhat-release ]; then
-            SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            if [ "${RHEL}" -eq 8 ]; then
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F ":" '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            else
+                SSL_VER_TMP=$(yum list installed|grep -i openssl|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+            fi
             export SSL_VER=".ssl${SSL_VER_TMP}"
```
Comment (EvgeniyPatlan) [inline build-ps/percona-server-5.7_builder.sh]: I have executed this code on centos7 and got  .ssl1:102

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: build-ps/percona-server-5.7_builder.sh (hunk 3 above)
Comment (Sudokamikaze) [reply build-ps/percona-server-5.7_builder.sh]: Fixed:
```
[root@compile-cent-7 mnt]# yum list installed|grep -i readline-devel|head -n1|awk '{print $2}'|awk -F "." '{print $1}'
6
```
```
[root@centos-6 vagrant]# yum list installed|grep -i readline-devel|head -n1|awk '{print $2}'|awk -F "." '{print $1}'
6
```

**Comment by hors** [comment]
> Percona-Server-5.7.29-32-Linux.x86_64.ssl11.readline8.tar.gz

Maybe it is better to use Percona-Server-5.7.29-32-Linux.x86_64.ssl11.deb.tar.gz and Percona-Server-5.7.29-32-Linux.x86_64.ssl11.rpm.tar.gz ?

**Comment by hors** [comment]
> > Percona-Server-5.7.29-32-Linux.x86_64.ssl11.readline8.tar.gz
> 
> Maybe it is better to use Percona-Server-5.7.29-32-Linux.x86_64.ssl11.deb.tar.gz and Percona-Server-5.7.29-32-Linux.x86_64.ssl11.rpm.tar.gz ?

We already have tow different  'ssl10[0|1].tar.gz' and 'ssl102.[deb|rpm].tar.gz' variants  and now we are adding one more.

**Comment by Sudokamikaze** [comment]
@hors 

There's a little problem:

As readline 8 is only in Ubuntu Focal(20.04), older DEB versions use readline 7.

And for RPM: Centos6/7 uses readline 6, Centos8 uses readline 8

**Comment by EvgeniyPatlan** [comment]
in general we are going to change UI and it might be confusing for our users. I think we should ask experts how to proceed.
I personally even dislike the idea of .deb and .rpm
the same about PXB tarballs.
Yes currently we have more dependencies and they are all different for all OS.
Maybe it is even high time to provide libs in tarballs.(just an idea)

**Comment by EvgeniyPatlan** [comment]
> in general we are going to change UI and it might be confusing for our users. I think we should ask experts how to proceed.
> I personally even dislike the idea of .deb and .rpm
> the same about PXB tarballs.
> Yes currently we have more dependencies and they are all different for all OS.
> Maybe it is even high time to provide libs in tarballs.(just an idea)

@Sudokamikaze please proceed with ^^^^

**Comment by Sudokamikaze** [comment]
> > in general we are going to change UI and it might be confusing for our users. I think we should ask experts how to proceed.
> > I personally even dislike the idea of .deb and .rpm
> > the same about PXB tarballs.
> > Yes currently we have more dependencies and they are all different for all OS.
> > Maybe it is even high time to provide libs in tarballs.(just an idea)
> 
> @Sudokamikaze please proceed with ^^^^

Closing MR due to latest decision of tarballs

**Pull request closed without merging**